initial_pos_x: [4,0,-2,2]
initial_pos_y: [2,2,-2,-2]
